python/huggingfaceserver/huggingfaceserver/model.py

```diff
@@ -88,7 +92,22 @@ def infer_task_from_model_architecture(model_config_path: str):
         if task is None:
             raise ValueError(f"Task couldn't be inferred from {architecture}. Please manually set `task` option.")
 
-    def load(self) -> bool:
+    @staticmethod
+    def infer_vllm_supported_from_model_architecture(model_config_path: str):
+        model_config = AutoConfig.from_pretrained(model_config_path)
+        architecture = model_config.architectures[0]
+        model_cls = ModelRegistry.load_model_cls(architecture)
+        if model_cls is None:
+            logger.info("not a supported model by vLLM")
+        return model_cls
+
+    def load(self, engine_args=None) -> bool:
+        if self.use_vllm and self.device == torch.device("cuda"):   # vllm needs gpu
```

Comment:
Do we need the `use_vllm` if we can determine the list of models architecture vllm can support?

Reply:
Thought going this route to set whether to use vLLM instead of using the global variable `_vllm`
`self.use_vllm = not kwargs.get('disable_vllm', False) if _vllm else False`

This is a pre-check if someone explicitly disables vLLM or does not have vLLM package in local.